sdk/identity/Azure.Identity/src/AuthenticationRecord.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Microsoft.Identity.Client;
+
+namespace Azure.Identity
+{
+
+
+    /// <summary>
+    /// Account information relating to an authentication request.
+    /// </summary>
+    public class AuthenticationRecord
+    {
+        private const string UsernamePropertyName = "username";
+        private const string AuthorityPropertyName = "authority";
+        private const string HomeAccountIdPropertyName = "homeAccountId";
+        private const string TenantIdPropertyName = "tenantId";
+
+        private static readonly JsonEncodedText s_usernamePropertyNameBytes = JsonEncodedText.Encode(UsernamePropertyName);
+        private static readonly JsonEncodedText s_authorityPropertyNameBytes = JsonEncodedText.Encode(AuthorityPropertyName);
+        private static readonly JsonEncodedText s_homeAccountIdPropertyNameBytes = JsonEncodedText.Encode(HomeAccountIdPropertyName);
+        private static readonly JsonEncodedText s_tenantIdPropertyNameBytes = JsonEncodedText.Encode(TenantIdPropertyName);
+
+        internal AuthenticationRecord()
+        {
+
+        }
+
+        internal AuthenticationRecord(AuthenticationResult authResult)
+        {
+            Username = authResult.Account.Username;
+            Authority = authResult.Account.Environment;
+            AccountId = authResult.Account.HomeAccountId;
+            TenantId = authResult.TenantId;
+        }
+
+        internal AuthenticationRecord(string username, string authority, string homeAccountId, string tenantId)
```

Comment:
Do you still want to add client id?

Reply:
I believe that we will probably need to but, I haven't added this yet here because it doesn't come back with the `AuthenticationResult` we get back from MSAL, so we'd need to wire it up in a different way then the rest of the data stored in `AuthenticationRecord`. I've created issue  https://github.com/Azure/azure-sdk-for-net/issues/11740 to track this.